src/ray/gcs/gcs_client/gcs_client.cc

```diff
@@ -147,13 +147,19 @@ std::pair<std::string, int> GcsClient::GetGcsServerAddress() const {
 }
 
 PythonGcsClient::PythonGcsClient(const GcsClientOptions &options) : options_(options) {
+  RAY_UNUSED(ConnectChannel());
+}
+
+Status PythonGcsClient::ConnectChannel() {
   channel_ =
       rpc::GcsRpcClient::CreateGcsChannel(options_.gcs_address_, options_.gcs_port_);
   kv_stub_ = rpc::InternalKVGcsService::NewStub(channel_);
   runtime_env_stub_ = rpc::RuntimeEnvGcsService::NewStub(channel_);
   node_info_stub_ = rpc::NodeInfoGcsService::NewStub(channel_);
   job_info_stub_ = rpc::JobInfoGcsService::NewStub(channel_);
   autoscaler_stub_ = rpc::autoscaler::AutoscalerStateService::NewStub(channel_);
+
+  return Status::OK();
```

Comment:
Is there any scenario that this func doesn't return OK status? Do we need some exception handling here if so?

Reply:
If the function never fail, we should change the return type to `void`

Reply:
This was the previous behavior. Anyway, since they are now combined, there are real chances of failure.